### PR TITLE
[MBEDTLS_PRIVATE] Add accessor for session and ciphersuite_id

### LINKED ISSUE #8529: Review requested `MBEDTLS_PRIVATE ` accessors.
### Suggested enhancement

As specified in the MBed-TLS 3.X migration guide [most structure fields are now private](https://github.com/Mbed-TLS/mbedtls/blob/909cf5a3ecc1e91c62e148f96f03400ca88869ff/docs/3.0-migration-guide.md#most-structure-fields-are-now-private) . Community can request the addition of private getter/setter functions for their use cases.

There has been a range of such requests #6151, #5441, #8456, #4383 , #7722 as well as some already planned as part of the transition work #5016, #5017

In some cases it may not be a straighforward ask e.g. #4383

The scope of this task is to collect all such requests under one design-review ticket and decide which one we will be implemented as part of the next relase, and which ones have special considerations.

### Scope
The output of this issue is a collection of s-sized subtasks, for accepted getters/setters, appropriately linked with the others items in the epic, and reviewed and approved by members of the team.

### PR DESCRIPTION
**Pull request by minosgalanakis**
## Description

The ps is adding accessors for the following members of the `mbedtls_ssl_session`
- `int MBEDTLS_PRIVATE(ciphersuite)` -> `mbedtls_ssl_session_get_ciphersuite_id()`
- `size_t MBEDTLS_PRIVATE(id_len)` -> `mbedtls_ssl_session_get_id_len()`
-`unsigned char MBEDTLS_PRIVATE(id)[32` -> `mbedtls_ssl_session_get_id()`

As well as one accessor for the `mbedtls_ssl_ciphersuite_t` struct
- `uint8_t MBEDTLS_PRIVATE(cipher)` -> `mbedtls_ssl_ciphersuite_get_id()`

Resolves #8529

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** Included
- [x] **backport** Not required new API calls.
- [x] **tests** provided

### CODE REVIEW

File: tests/suites/test_suite_ssl.function [hunk 1]
```diff
@@ -2379,6 +2379,54 @@ exit:
...
+            break;
+    }
+    TEST_ASSERT(*mbedtls_ssl_session_get_id(&session) == session.id);
+    TEST_ASSERT(mbedtls_ssl_session_get_id_len(&session) == session.id_len);
+    /* mbedtls_test_ssl_tls1x_populate_session sets a mock suite-id of 0xabcd */
+    TEST_ASSERT(mbedtls_ssl_session_get_ciphersuite_id(&session) == 0xabcd);
+
+    /* Test setting a reference id for tls1.3 and tls1.2 */
+    ciphersuite_info = mbedtls_ssl_ciphersuite_from_id(ciphersuite_id);
+    if (ciphersuite_info != NULL) {
```
Comment (minosgalanakis) [inline tests/suites/test_suite_ssl.function:2420]: In certain configurations the `ssl_ciphersuite_from_id` will return NULL. Testing this function is outside of the scope so we only test the getter for the other cases where it will create a `ciphersuite_info` stucture.

File: ChangeLog.d/add_ssl_session_accessors.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Features
+   * Add new accessors to expose the private session-id,
+     session-id length, and ciphersuite-id members of
+     `mbedtls_ecdh_conmbedtls_ssl_sessiontext` structure.
```
Comment (davidhorstmann-arm) [inline ChangeLog.d/add_ssl_session_accessors.txt:4]: Copypasta typo, perhaps?
```suggestion
     `mbedtls_ssl_session` structure.
```

File: ChangeLog.d/add_ssl_session_accessors.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+     Add new accessor to expose the ciphersuite-id of
+     `mbedtls_ssl_ciphersuite` structure.Design ref: #8529
```
Comment (davidhorstmann-arm) [inline ChangeLog.d/add_ssl_session_accessors.txt:6]: Nit:
```suggestion
     `mbedtls_ssl_ciphersuite_t` structure.Design ref: #8529
```

**Review by davidhorstmann-arm — CHANGES_REQUESTED** [review]
One typo, LGTM otherwise

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Review by tom-daubney-arm — CHANGES_REQUESTED** [review]
Couple of small typos but otherwise it looks good.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!
